tests/unit/test_charm_non_leader_unit.py

```diff
@@ -0,0 +1,224 @@
+# Copyright 2022 Canonical Ltd.
+# See LICENSE file for licensing details.
+
+import unittest
+from unittest.mock import Mock, PropertyMock, call, patch
+
+from ops import testing
+from ops.model import ActiveStatus, BlockedStatus, WaitingStatus
+
+from charm import AlertmanagerConfigurerOperatorCharm
+
+testing.SIMULATE_CAN_CONNECT = True
+
+TEST_MULTITENANT_LABEL = "some_test_label"
+TEST_CONFIG = f"""options:
+  multitenant_label:
+    type: string
+    description: |
+      Alertmanager Configurer has been designed to support multiple tenants. In a multitenant
+      Alertmanager Configurer setup, each alert is first routed on the tenancy label, and then
+      the routing tree is distinct for each tenant.
+    default: {TEST_MULTITENANT_LABEL}
+"""
+with open("./tests/unit/test_config/alertmanager_default.yml", "r") as default_yaml:
+    TEST_ALERTMANAGER_DEFAULT_CONFIG = default_yaml.read()
+TEST_ALERTMANAGER_CONFIG_FILE = "/test/rules/dir/config_file.yml"
+ALERTMANAGER_CLASS = "charm.AlertmanagerConfigurerOperatorCharm"
+
+
+class TestAlertmanagerConfigurerOperatorCharmNonLeader(unittest.TestCase):
+    @patch("charm.KubernetesServicePatch", lambda charm, ports: None)
+    def setUp(self):
+        testing.SIMULATE_CAN_CONNECT = True
+        self.harness = testing.Harness(AlertmanagerConfigurerOperatorCharm, config=TEST_CONFIG)
+        self.addCleanup(self.harness.cleanup)
+        self.harness.begin()
+
+    @patch("charm.AlertmanagerConfigDirWatcher")
+    @patch(f"{ALERTMANAGER_CLASS}.ALERTMANAGER_CONFIG_DIR", new_callable=PropertyMock)
+    @patch("ops.model.Container.push", Mock())
+    def test_given_alertmanager_config_directory_when_start_then_watchdog_starts_watching_given_alertmanager_config_directory(  # noqa: E501
+        self, patched_config_dir, patched_alertmanager_config_dir_watcher
+    ):
+        test_config_dir = "/test/rules/dir"
+        patched_config_dir.return_value = test_config_dir
+        self.harness.charm.on.start.emit()
+
+        patched_alertmanager_config_dir_watcher.assert_called_with(
+            self.harness.charm, test_config_dir
+        )
+
+    @patch("charm.AlertmanagerConfigDirWatcher", Mock())
+    def test_given_alertmanager_relation_not_created_when_pebble_ready_then_charm_goes_to_blocked_state(  # noqa: E501
+        self,
+    ):
+        self.harness.container_pebble_ready("alertmanager-configurer")
+
+        assert self.harness.charm.unit.status == BlockedStatus(
+            "Waiting for alertmanager relation to be created"
+        )
+
+    @patch("charm.AlertmanagerConfigDirWatcher", Mock())
+    def test_given_alertmanager_relation_created_but_alertmanager_configurer_container_not_yet_ready_when_pebble_ready_then_charm_goes_to_waiting_state(  # noqa: E501
+        self,
+    ):
+        self.harness.add_relation("alertmanager", "alertmanager-k8s")
+        testing.SIMULATE_CAN_CONNECT = False
+        self.harness.container_pebble_ready("alertmanager-configurer")
```

Comment:
"alertmanager-configurer" is refered to quite often, should probably be a variable

Reply:
Replaced with `self.harness.charm.ALERTMANAGER_CONFIGURER_SERVICE_NAME`. 
It will keep the tests in sync with the charm implementation.